portfolio/src/main/webapp/script.js

```diff
@@ -42,3 +42,515 @@ function visitLink(s){
+                    square.col == this.selected_square.col - 1) &&
+                    square.piece){
+                    
+                    selected_piece.moved = true;
+                    return true;
+                }
+                else{
+                    return false;
+                }
+            }
+            else{
+                if(square.row == this.selected_square.row + 1 && 
+                    square.col == this.selected_square.col && !square.piece){
+
+                    selected_piece.moved = true;
+                    return true;
+                }
+                else if(square.row == this.selected_square.row + 2 && 
+                    square.col == this.selected_square.col && !selected_piece.moved && 
+                    !square.piece){
+
+                    selected_piece.moved = true;
+                    return true;
+                }
+                else if(square.row == this.selected_square.row + 1 && 
+                    (square.col == this.selected_square.col + 1 || 
+                    square.col == this.selected_square.col - 1) &&
+                    square.piece){
+                    
+                    selected_piece.moved = true;
+                    return true;
+                }
+                else{
+                    return false;
+                }
+            }
+        }
+        else if(selected_piece.type == 'knight'){
+            if(square.row == this.selected_square.row - 2 && 
+                square.col == this.selected_square.col - 1){
+
+                return true;
+            }
+            else if(square.row == this.selected_square.row - 2 && 
+                square.col == this.selected_square.col + 1){
+                
+                return true;
+            }
+            else if(square.row == this.selected_square.row - 1 && 
+                square.col == this.selected_square.col + 2){
+
+                return true;
+            }
+            else if(square.row == this.selected_square.row + 1 && 
+                square.col == this.selected_square.col + 2){
+
+                return true;
+            }
+            else if(square.row == this.selected_square.row + 2 && 
+                square.col == this.selected_square.col + 1){
+
+                return true;
+            }
+            else if(square.row == this.selected_square.row + 2 && 
+                square.col == this.selected_square.col - 1){
+
+                return true;
+            }
+            else if(square.row == this.selected_square.row + 1 && 
+                square.col == this.selected_square.col - 2){
+
+                return true;
+            }
+            else if(square.row == this.selected_square.row - 1 && 
+                square.col == this.selected_square.col - 2){
+
+                return true;
+            }
+            else{
+                return false;
+            }
+        }
+        else if(selected_piece.type == 'bishop'){
```

Comment:
There actually is a catch for if the piece doesn't move, and that's if the new square has a selected piece of the same color it will select that piece and not move the first piece.

Reply:
I will comment this